From 7f32e5fc410a6295a7ddbbf6eda62a577d7b940b Mon Sep 17 00:00:00 2001
From: anatasluo <luolongjuna@gmail.com>
Date: Tue, 8 Mar 2022 08:34:14 +0000
Subject: [PATCH 139/249] proc_smaps: remove useless nonlinear check

nonlinear information has been removed since commit
1da4b35b001481df99 in kernel.

Signed-off-by: anatasluo <luolongjuna@gmail.com>
---
 criu/proc_parse.c | 12 +-----------
 1 file changed, 1 insertion(+), 11 deletions(-)

diff --git a/criu/proc_parse.c b/criu/proc_parse.c
index 13ec76e3b..5cd5e6db8 100644
--- a/criu/proc_parse.c
+++ b/criu/proc_parse.c
@@ -754,17 +754,7 @@ int parse_smaps(pid_t pid, struct vm_area_list *vma_area_list, dump_filemap_t du
 		eof = (str == NULL);
 
 		if (!eof && !__is_vma_range_fmt(str)) {
-			if (!strncmp(str, "Nonlinear", 9)) {
-				BUG_ON(!vma_area);
-				pr_err("Nonlinear mapping found %016" PRIx64 "-%016" PRIx64 "\n", vma_area->e->start,
-				       vma_area->e->end);
-				/*
-				 * VMA is already on list and will be
-				 * freed later as list get destroyed.
-				 */
-				vma_area = NULL;
-				goto err;
-			} else if (!strncmp(str, "VmFlags: ", 9)) {
+			if (!strncmp(str, "VmFlags: ", 9)) {
 				BUG_ON(!vma_area);
 				parse_vma_vmflags(&str[9], vma_area);
 				continue;
-- 
2.35.1

